mlflow/genai/evaluation/harness.py

```diff
@@ -124,6 +124,14 @@ def _run_single(
     assessments.extend(eval_item.get_expectation_assessments())
     eval_result = EvalResult(eval_item=eval_item, assessments=assessments)
 
+    tags = eval_item.tags or {}
+    for key, value in tags.items():
+        try:
+            mlflow.set_tag(key, value)
```

Comment:
Any chance that a non-string key or value will break things? (Can we add guards?)